### PR TITLE
12307 Update business-create-ui to use legal API to V2

### PR DESCRIPTION
**Pull request by lewischenstudio**
*Issue #:* [/bcgov/entity#12307](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/12307)

*Description of changes:*
- renamed url in `legal.services.ts`
- renamed `legalApiUrl` in `config-helper.ts`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "business-create-ui",
-  "version": "4.1.1",
+  "version": "4.1.2",
```
Comment (severinbeauvais) [inline package.json:3]: Another PR was just merged with this version. Please update.

**Review by severinbeauvais — COMMENTED** [review]
Did you test that all existing Legal API calls (ie, any call that uses Base URL) still function correctly?

File: src/store/getters/state-getters.ts [hunk 2]
```diff
@@ -468,6 +468,7 @@ export const isDissolutionValid = (state: StateIF): boolean => {
   const isCourtOrderValid = isRoleStaff(state) ? getCourtOrderStep(state).valid : true
   const isStaffPaymentValid = isRoleStaff(state) ? getStaffPaymentStep(state).valid : true
 
+  // eslint-disable-next-line
```
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:471]: Why not split the line instead?

Also, you should disable all lint for a line; only the error you want to disable.

Also, Megan is updating this anyway: https://github.com/bcgov/business-create-ui/pull/422#discussion_r917089556

Comment (severinbeauvais) [reply src/store/getters/state-getters.ts:471]: ^^ What I would do is make this multi-line in this PR, and then if Megan's PR goes first then you can remove it. If Megan's PR goes second, she can remove it.

**Comment by severinbeauvais** [comment]
> Did you test that all existing Legal API calls (ie, any call that uses Base URL) still function correctly?

Specifically, all Legal API calls except the subject one will now use V2. Do they still work as before?

**Comment by lewischenstudio** [comment]
> > Did you test that all existing Legal API calls (ie, any call that uses Base URL) still function correctly?
> 
> Specifically, all Legal API calls except the subject one will now use V2. Do they still work as before?

Yes. Endpoints using legal-api V2 are working as they before.

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**